src/opentimelineio/composition.cpp

```diff
@@ -335,4 +335,167 @@ bool Composition::has_child(Composable* child) const {
     return _child_set.find(child) != _child_set.end();
 }
 
+SerializableObject::Retainer<Composable> Composition::child_at_time(
+    RationalTime const& search_time,
+    ErrorStatus* error_status,
+    bool shallow_search) const
+{
+    auto range_map = range_of_all_children(error_status);
+    if (!error_status) {
```

Comment:
Hi, I didn't test the code but the if branch clearly leads to segmentation fault. The `!` should be removed otherwise a null pointer gets dereferenced. This is repeated also in following lines of code.

Reply:
OK, I've created a new issue #1060, and PR #1061, to fix checking the error_status. I did not add checks for whether the error_status pointer is valid, since that doesn't happen in the surrounding code.